src/sentry/spans/consumers/process/factory.py

```diff
@@ -0,0 +1,44 @@
+from typing import Mapping
+
+import msgpack
+from arroyo.backends.kafka.consumer import KafkaPayload
+from arroyo.processing.strategies.abstract import ProcessingStrategy, ProcessingStrategyFactory
+from arroyo.processing.strategies.commit import CommitOffsets
+from arroyo.processing.strategies.produce import Produce
+from arroyo.processing.strategies.run_task import RunTask
+from arroyo.types import Commit, Message, Partition, Topic
+from confluent_kafka import Producer
+
+from sentry.spans.span import Span
+from sentry.utils import json, kafka_config
+
+
+def process_message(message: Message[KafkaPayload]) -> None:
+    payload = msgpack.unpackb(message.payload.value)
+    span_dict = json.loads(payload["span"])
```

Comment:
The span object is now pure msgpack (no json encoding), for example

```
b'\x84\xa4type\xa4span\xa4span\x87\xabdescription\xd9"GET /api/0/organizations/?member=1\xa2op\xa4http\xaeparent_span_id\xb0aaaaaaaaaaaaaaaa\xa7span_id\xb0bbbbbbbbbbbbbbbb\xafstart_timestamp\xcb@\x8f@\x00\x00\x00\x00\x00\xa9timestamp\xcb@\xa7p\x00\x00\x00\x00\x00\xa8trace_id\xd9 ff62a8b040f340bda5d830223def1d81\xaastart_time\xced\xcc\xf6E\xaaproject_id*'
```

See 
 https://github.com/getsentry/relay/blob/c7300d8c58460eff312695889ce0eb1bb7c509cf/tests/integration/test_store.py#L1209-L1213

Reply:
@untitaker , @lynnagara  did we support msgpack in the schema registry in the end ?

Reply:
Is there a specific reason for using msgpack and not JSON? I understand the benefits of msgpack. But would like to avoid heterogeneous message formats in the pipeline if possible.

Reply:
msgpack is supported in the schema registry. the schema is still jsonschema, and bytestrings are typed as `Any`